Tests/Opc.Ua.Security.Certificates.Tests/CertificateTestsForRSA.cs

```diff
@@ -333,182 +336,201 @@ public void CreateRSAManualSerialTest()
-                using (var appCert = CertificateBuilder.Create("CN=App Cert")
-                    .SetIssuer(issuer)
-                    .CreateForRSA(generator))
-                {
-                    Assert.NotNull(appCert);
-                    WriteCertificate(appCert, "Signed RSA app cert");
-                    CheckPEMWriter(appCert);
-                }
+            {
+                Assert.NotNull(cert);
+                issuer = X509CertificateLoader.LoadCertificate(cert.RawData);
+                WriteCertificate(cert, "Default root cert with supplied RSA cert");
+                CheckPEMWriter(cert);
             }
+
+            // now sign a cert with supplied private key
+            using X509Certificate2 appCert = CertificateBuilder
+                .Create("CN=App Cert")
+                .SetIssuer(issuer)
+                .CreateForRSA(generator);
+            Assert.NotNull(appCert);
+            WriteCertificate(appCert, "Signed RSA app cert");
+            CheckPEMWriter(appCert);
         }
 
 #if NETFRAMEWORK || NETCOREAPP3_1_OR_GREATER
         [Test]
-        [SuppressMessage("Interoperability", "CA1416: Validate platform compatibility", Justification = "Test is ignored.")]
         public void CreateIssuerRSACngWithSuppliedKeyPair()
         {
             if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                Assert.Ignore("Cng provider only available on windows");
+                NUnit.Framework.Assert.Ignore("Cng provider only available on windows");
             }
             X509Certificate2 issuer = null;
-            CngKey cngKey = CngKey.Create(CngAlgorithm.Rsa);
-            using (RSA rsaKeyPair = new RSACng(cngKey))
-            {
-                // create cert with supplied keys
-                var generator = X509SignatureGenerator.CreateForRSA(rsaKeyPair, RSASignaturePadding.Pkcs1);
-                using (var cert = CertificateBuilder.Create("CN=Root Cert")
+#pragma warning disable IDE0079 // Remove unnecessary suppression
+#pragma warning disable CA1416 // Validate platform compatibility
+#pragma warning restore IDE0079 // Remove unnecessary suppression
+            var cngKey = CngKey.Create(CngAlgorithm.Rsa);
+#pragma warning disable IDE0079 // Remove unnecessary suppression
+#pragma warning restore CA1416 // Validate platform compatibility
+#pragma warning restore IDE0079 // Remove unnecessary suppression
+#pragma warning disable IDE0079 // Remove unnecessary suppression
+#pragma warning disable CA1416 // Validate platform compatibility
+#pragma warning restore IDE0079 // Remove unnecessary suppression
+            using RSA rsaKeyPair = new RSACng(cngKey);
+#pragma warning disable IDE0079 // Remove unnecessary suppression
+#pragma warning restore CA1416 // Validate platform compatibility
+#pragma warning restore IDE0079 // Remove unnecessary suppression
+            // create cert with supplied keys
```

Comment:
Why do we have so many duplicate pragma directives?

Reply:
trying to coax vs2022 to remove suggestions from the error list.  Also apparently restore cannot be merged, so this was the only way it dropped of the list.